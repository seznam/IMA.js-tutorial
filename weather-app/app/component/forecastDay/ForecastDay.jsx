import { AbstractPureComponent } from '@ima/core';
import React from 'react';
import PropTypes from 'prop-types';

import ForecastIcon from 'app/component/forecastIcon/ForecastIcon';

import PrecipType from 'app/constant/PrecipType';
import ForecastIconType from 'app/constant/ForecastIconType';

export default class ForecastDay extends AbstractPureComponent {
  static get propTypes() {
    return {
      forecast: PropTypes.shape({
        localDate: PropTypes.string,
        bio: PropTypes.number,
        icon: PropTypes.oneOf(Object.values(ForecastIconType)),
        isStorm: PropTypes.bool,
        precip: PropTypes.number,
        precipType: PropTypes.oneOf(Object.values(PrecipType)),
        pressureMax: PropTypes.number,
        pressureMin: PropTypes.number,
        rhMax: PropTypes.number,
        rhMin: PropTypes.number,
        snowPrecip: PropTypes.number,
        summary: PropTypes.string,
        sunrise: PropTypes.string,
        sunset: PropTypes.string,
        tempAvg: PropTypes.number,
        tempMax: PropTypes.number,
        tempMin: PropTypes.number,
        weekday: PropTypes.number,
        wind: PropTypes.number,
        windDir: PropTypes.number
      }).isRequired,
      place: PropTypes.shape({
        TZoffsetFormatted: PropTypes.string
      }),
      isActive: PropTypes.bool,
      onClick: PropTypes.func
    };
  }

  get _forecastTypeCSSClass() {
    return '';
  }

  render() {
    const { forecast, isActive, onClick } = this.props;
    const date = new Date(forecast.localDate);

    return (
      <div
        className={this.cssClasses({
          'forecast-day': true,
          'forecast-day--active': isActive,
          [`forecast-day--${this._forecastTypeCSSClass}`]: true
        })}
        onClick={event => onClick(event)}>
        <h3 className={this.cssClasses('forecast-day__date')}>
          {`${date.getDate()}. ${date.getMonth() + 1}.`}
        </h3>
        <span className={this.cssClasses('forecast-day__temp')}>
          {Math.round(forecast.tempMin)} - {Math.round(forecast.tempMax)} Â°C
        </span>
        <div className={this.cssClasses('forecast-day__sunrise-sunset')}>
          <span>{forecast.sunrise}</span>&#8212;<span>{forecast.sunset}</span>
        </div>
        <div className={this.cssClasses('forecast-day__icon')}>
          {this._renderIcon()}
        </div>
      </div>
    );
  }

  _renderIcon() {
    const { forecast } = this.props;

    return <ForecastIcon icon={forecast.icon} isDay={true} />;
  }
}
