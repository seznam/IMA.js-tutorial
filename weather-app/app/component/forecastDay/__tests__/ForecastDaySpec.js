import React from 'react';
import { defaultCssClasses } from '@ima/core';
import { shallow } from 'enzyme';
import { withContext } from 'shallow-with-context';
import ForecastDay from '../ForecastDay';

describe('ForecastDay', () => {
    it('can be rendered', () => {
        const props = {
            forecast: {
                localDate: '2019-03-18T08:06:08.341Z',
                tempMin: 5,
                tempMax: 10,
                sunrise: '6:00',
                sunset: '20:00',
                icon: 0
            }
        };
        const context = {
            $Utils: {
                $CssClasses: defaultCssClasses
            }
        };
        const ForecastDayWithContext = withContext(ForecastDay, context);
        const wrapper = shallow(<ForecastDayWithContext {...props} />, { context });

        expect(wrapper).toMatchSnapshot();
    });
});
