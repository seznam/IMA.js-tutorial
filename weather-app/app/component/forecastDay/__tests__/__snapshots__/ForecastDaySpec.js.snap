// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDay can be rendered 1`] = `
<div
  className="forecast-day forecast-day--"
  onClick={[Function]}
>
  <h3
    className="forecast-day__date"
  >
    18. 3.
  </h3>
  <span
    className="forecast-day__temp"
  >
    5
     - 
    10
     °C
  </span>
  <div
    className="forecast-day__sunrise-sunset"
  >
    <span>
      6:00
    </span>
    —
    <span>
      20:00
    </span>
  </div>
  <div
    className="forecast-day__icon"
  >
    <ForecastDay
      icon={0}
      isDay={true}
    />
  </div>
</div>
`;
